### ISSUE TITLE
User story - Associate company with Job Post

### ISSUE THREAD

**Issue by stephendawsondev** [stephendawsondev/StackPortfolio#38]
### If applicable, add a related Epic.

Epic:  #28 

### User Story

As a **recruiter user**, I can **link my job post to a company** so that I can **add authenticity to the role**.

### Acceptance Criteria

- Recruiter user can choose a company from a list of companies or...
- Have the option to add a new company
- If the company is added, it will need to be approved first, so should appear greyed out

### Please ensure you've completed these tasks before marking the issue as done.

- [ ] Reviewed the user story for clarity and completeness.
- [ ] Identified and linked related issues or epics.
- [ ] Created or updated the necessary documentation.
- [ ] Written unit and/or integration tests for functionality.
- [ ] All code changes have been manually tested to ensure they meet the acceptance criteria.
- [ ] Updated the issue with any relevant notes or feedback (to date).
- [ ] Merged into the main branch and deployed to production (if applicable).
